### DISCUSSION TITLE
Permission denied in airflow official docker image for apt-get

### DISCUSSION THREAD

**Discussion by gorun2git** [apache/airflow#17766] [Q&A] [answered]
Hi all,
I'm trying to deploy airflow on the AWS EKS cluster using the official airflow helm chart. The chart is using docker image **apache/airflow:2.1.2**. Also, I'm using default airflow user with **uid:50000** and **gid:0.** Apparently, the user can't execute **apt-get** commands. I've also tried to try sudo with the 'airflow' password but no luck.

Actually, I want to create a custom image from this image and install some additional packages on it. The Dockerfile is like

> FROM apache/airflow:2.1.2
> 
> RUN apt-get update \
>     && apt-get install -y --no-install-recommends vim curl git rsync unzip \
>     && apt-get autoremove -y \
>     && apt-get clean

Apparently, it's using default airflow user, my question is can I somehow get sudo rights (for sure I've missed the password for the airflow user), or should I in my Dockerfile create another user and continue with it? I wonder what is the best practice? 

Thank you in advance.

**Comment by potiuk** [comment, marked as answer]
This is expected. Airflow user is "regular" user that has no sudoers capability by default. You can build your image in the way to add airflow to become sudoer, but this is not really needed. If you build your image and want to use apt-get or any other root capability you should temporarily switch to "root" use and switch back to airflow user after you are done:

```
FROM apache/airflow:2.1.2

USER root

RUN apt-get update
&& apt-get install -y --no-install-recommends vim curl git rsync unzip
&& apt-get autoremove -y
&& apt-get clean

USER airflow
```

I highly recommend the documentation of "How to Build your image" on Airflow website. It has pretty comprehensive explanation and plenty of examples (including apt example): https://airflow.apache.org/docs/docker-stack/build.html#adding-new-apt-package - it also explains a lot of why's and gives some helpful hints on how you can build a highly optimised image.

**Reply by gorun2git** [reply]
Thank you @potiuk 🙏  I'll definitely check it.
